### DISCUSSION TITLE
"No map data" on Viomi V8

### DISCUSSION THREAD

**Discussion by DanielsSt** [Hypfer/Valetudo#1711] [Q&A / Support] [answered]
Hello everyone!
Looking for ideas where else to look / what else to try.
I doubt that someone will provide working solution, since I most likely am the one who messed it up in the first place, but would appreciate some hints where to look because I cannot find overall explanation how this setup is expected to be working and what could be causing the issue.

The issue, as subject says - "No map data" default map is showing up.
Persistent map is turned on, and everything else is working fine, I can control robot, change fan speed etc, but whatever I do map is stuck on "no map data"

Anyway... I believe knowing the whole story might give you some insight (and because of some of the steps I am not putting too much hope in this thread)
So... from the beginning:

- At first got ssh/root access to my Viomi V8 robot, yay
  - followed these instructions - https://github.com/rumpeltux/viomi-rooting
- Got Valetudo installed (the script I was using installed version 2021.03.0)
- Everything seemed to be working fine, but since the version looked... mildly saying ancient, went ahead and installed the latest version (2023.01.0)
  - Confirmed that everything works and already started to celebrate my success 🎉 
- One day later thought that it would be funny to get `librespot` (Spotify client) working on it (_just because I can, not like I needed it_ 😅 ), managed to compile it for armv7 and pushed it robot
  - Robot was missing libs from `arm-linux-gnueabihf` that were needed for librespot, pushed them over, symlinked them and it was at least starting, probably if I built it differently with musli-dev it might have been different, who know unfortunately I ain't no C / rust dev 😓 
  - Added Mozilla CA certificates in `/etc/ssl/certs.pem` which were also needed for thing to start
  - Then kept tinkering with ALSA in hopes to actually get sound out of it.
  - During the process restarted robot a couple of times from SSH and noticed that Valetudo is showing default "No map data" map. Not sure if it is causation or correlation, but nevertheless, worth mentioning.
  - Since I could not get ALSA working properly, yeet'ed files I had transferred to robot (my guess ALSA on the robot is too  old, but the topic is not about it 😆)
- I hoped that the issue will self-heal after doing full cleanup, but that did not help, tried twice
- Decided that if I have broken it, I might as well flash custom firmware, went to [DustBuilder](https://dustbuilder.dontvacuum.me/_viomi_v7.html) to build image for it (the one that does conversion to V6, from Valetudo docs came to a conclusion that capabilities are matching)
  - Flashed firmware
  - Yes, I am reckless and chose not to take backups from `/dev/nand*`
  - Fixed Wifi which was reset after flashing firmware by editing `wpa_supplicant.conf`
- Still Valetudo greeted me with "No map data" map :) (and now showing up as V6, but that was expected)
- Took a look into this rooting instruction https://github.com/Hypfer/valetudo-crl200s-root since it mentions usage of dustbuilders firmware
 - Started from end of 5th step, by running `./convert_robot.sh` 
   - No output from script tho, which should indicate that nothing was missing
   - re-flashed Dustbuilders firmware for a good measure
- Still "No map data"
- Started searching for similar cases on internet
   - this issue looked promising - https://github.com/Hypfer/Valetudo/issues/1170
     - tried changing values on `/mnt/UDISK/config/sysConfig.ini` (it didn't solve map data issue)
       - server_cmd_port=8053
       - server_map_port=8079
       - although, changes were not persistent after reboot, added some `sed` action on `/mnt/UDISK/_root.sh`
       - updated hosts file since it didn't include `3irobotics.net` domains mentioned in the ini file
     - checked `pool.ntp.org` accessibility, since it was pointed as possible culprit (ping worked)
       - Also on Valetudo logs could see occasional record about successfully setting time
- Tried resetting the map / toggling map persistence setting, but that did not change the situation
- Tried to figure out where map data could be, but I guess it is stored somewhere in format that did not ring a bell
- Upgraded Valetudo to 2023.02.0
- Decided to spend Monday evening by sharing how adventurous weekend I had in hopes any of you might have some wild (or not so wild) ideas

Adding log with TRACE level - https://pastebin.com/4Sad52Ya
During logging session - opened Valetudo, toggled Persistent map setting off / on and started robot for a few moments (I remember that when it worked it would start redrawing map almost instantly)

**Comment by Hypfer** [comment, marked as answer]
> Fixed Wifi which was reset after flashing firmware by editing wpa_supplicant.conf

While this did connect the OS to your Wi-Fi network, as far as the firmware itself is concerned, you're still unprovisioned, meaning that it won't connect to the mock cloud and thus also not upload any maps.

To fix this, just press and hold the two buttons (Power, Home) until it starts talking about a reset of Wi-Fi credentials.
Then, connect to the Wi-Fi AP of the robot and use the Valetudo webinterface to configure Wi-Fi as described here: https://valetudo.cloud/pages/general/getting-started.html#joining_wifi

**Reply by DanielsSt** [reply]
From all the things that might have gone wrong I am glad it was only Wifi issue.
Thanks a lot, reset of Wifi credentials helped, now map-related things are working as expected 🎉
